Frontend/CO.CDP.OrganisationApp/Pages/Supplier/ConnectedEntity/ConnectedEntityIndividualCategory.cshtml.cs

```diff
@@ -40,7 +39,15 @@ public IActionResult OnGet()
     public IActionResult OnPost()
     {
         var state = session.Get<ConnectedEntityState>(Session.ConnectedPersonKey);
-        if (state == null || !ModelState.IsValid)
+
+        if (state == null)
+        {
+            return RedirectToPage("ConnectedEntitySupplierCompanyQuestion", new { Id });
+        }
+
+        ConnectedEntityType = state.ConnectedEntityType;
```

Comment:
This is the crux of the fix - `ConnectedEntityType` was not being assigned a value on post, so when it re-rendered the page because of validation errors, you were being presented with the options of trustees instead of individuals and so it would get itself in a loop, redirecting you to "here" because it would not flow through `GetRedirectLinkPageName` correctly